### PR TITLE
Sc msg cert removal

### PR DESCRIPTION
**Pull request by alsala**
Removed MSG_CERT which was added to network protocol for handling certificates and made MSG_TX message able to handle both transactions and certificates.

### CODE REVIEW

File: src/primitives/transaction.h [hunk 1]
```diff
@@ -705,7 +705,8 @@ class CTransactionBase
         return !(a==b);
     }
 
-    virtual bool IsScVersion() const = 0;
+    virtual bool IsCertificateVersion() const { return false; }
```
Comment (abi87) [inline src/primitives/transaction.h]: The method `IsCertificateVersion` is never called on a CTransaction. For ScCertificate it is called only once, in `Sidechain::checkCertSemanticValidity(const CScCertificate& cert, CValidationState& state)`
I would remove this method altogether and explicitly compare the versions in 'Sidechain::checkCertSemanticValidity(const CScCertificate& cert, CValidationState& state)'

File: src/primitives/transaction.h [hunk 2]
```diff
@@ -705,7 +705,8 @@ class CTransactionBase
...
+    virtual bool IsScVersion() const { return false; }
```
Comment (abi87) [inline src/primitives/transaction.h]: `IsScVersion` is called only on CTransaction, never on CTransactionBase.
I would Remove it for CTransactionBase. Maybe also rename to IsScRelated

**Review by abi87 — CHANGES_REQUESTED** [review]

File: src/primitives/transaction.h (hunk 1 above)
Comment (alsala) [reply src/primitives/transaction.h]: Ok

File: src/primitives/transaction.h (hunk 2 above)
Comment (alsala) [reply src/primitives/transaction.h]: Ok, but for now I leave the name as it is

**Pull request merged**